src/components/Image/index.native.js

```diff
@@ -18,7 +18,10 @@ function Image(props) {
     const {source, isAuthTokenRequired, session, ...rest} = props;
 
     let imageSource = source;
-    if (typeof source !== 'number' && isAuthTokenRequired) {
+    if (source && source.uri && typeof source.uri === 'number') {
+        imageSource = source.uri;
```

Comment:
@parasharrajat 

could you test other Image components used in App to make sure this doesn't break other parts of App?